tidy3d/components/data/sim_data.py

```diff
@@ -228,26 +318,79 @@ def get_intensity(self, field_monitor_name: str) -> xr.DataArray:
             Data is interpolated to the center locations on Yee grid.
         """
 
-        field_dataset = self.at_centers(field_monitor_name)
+        if field_name[0] == "S":
+            dataset = self.get_poynting_vector(field_monitor_name)
+            if len(field_name) > 1:
+                if field_name in dataset:
+                    derived_data = dataset[field_name]
+                    derived_data.name = field_name
+                    return self._field_component_value(derived_data, val)
+                raise Tidy3dKeyError(f"Poynting component {field_name} not available")
+        else:
+            dataset = self.at_centers(field_monitor_name)
 
-        field_components = ("Ex", "Ey", "Ez")
-        if not all(field_cmp in field_dataset for field_cmp in field_components):
-            raise DataError(
-                "Field monitor must contain 'Ex', 'Ey', and 'Ez' fields to compute intensity."
-            )
+        if field_name in ("E", "H", "S"):
+            # Gather vector components
+            required_components = [field_name + c for c in "xyz"]
+            if not all(field_cmp in dataset for field_cmp in required_components):
+                raise DataError(
+                    f"Field monitor must contain '{field_name}x', '{field_name}y', and "
+                    f"'{field_name}z' fields to compute '{field_name}."
+                )
+            field_components = (dataset[c] for c in required_components)
+
+            # Apply the requested transformation
+            if val == "real":
+                derived_data = sum(f.real**2 for f in field_components) ** 0.5
+                derived_data.name = f"|Re{{{field_name}}}|"
```

Comment:
Related to Tyler's comment, maybe this could behave differently depending on the number of components requested? E.g. if only one is requested, then just return `real(thing)` to preserve the sign, otherwise do what you have right now?

Reply:
Yes, this happens in [`_field_component_value`](https://github.com/flexcompute/tidy3d/blob/f03b372cf5a5dba345b50040492a9ab4489ad45c/tidy3d/components/data/sim_data.py#L265). As I mentined in the comment to Tyler, `get_derived_field` always returns scalar values (for plotting, originally), that's why this happens.